packages/icons/processIcons.ts

```diff
@@ -0,0 +1,141 @@
+import fs from 'fs';
```

Comment:
This is the file that does the brunt of the work in this PR, I've done my best to split it up into logical sections and add comments describing what's happening at each step. Happy to go over it with someone though if that's easier!